### PR TITLE
Add a right click context menu to the systray applet (to "quit/exit" it and run arch-update)

### LINKED ISSUE #163: Add a right click menu to the systray applet to "quit/exit" it
It would be great to have a right click menu for the systray applet to *gracefully* quit/close it (instead of requiring people to either manually stop the systemd service or killing the process when needing to re-launch the systray from the terminal to get some logs for instance).

That would ideally imply introducing the translation code so we can have the menu entries in multiple languages.

While we're at it, I guess an entry to actually launch arch-update in the said menu wouldn't hurt :)

### PR DESCRIPTION
**Pull request by trigg**
- Add gettext translation to systray
- Add right-click-menu with two options
- - Run Arch-Update
- - Exit
- Added blank translation keys
Fixes #163

### CODE REVIEW

**Comment by Antiz96** [comment]
Thanks! I'll review/test that later today after work ;)

**Comment by Antiz96** [comment]
Thanks!

**Comment by Antiz96** [comment]
I know this is unrelated to that PR, but is it fine if we change the name of the function that runs Arch-Update from "update" to "run", or something?

I know this is a minor detail, but I think I would prefer the name of that function to reflect better what it is actually doing (namely running Arch-Update) rather than "update" (which is *one of* the functions of Arch-Update). What do you think?

File: po/arch-update.pot [hunk 1]
```diff
@@ -457,3 +457,11 @@ msgstr ""
 #, sh-format
 msgid "The '${config_file}' configuration file has been generated"
 msgstr ""
+
+#: src/script/arch-update-tray.py:111
+msgid "Update"
```
Comment (Antiz96) [inline po/arch-update.pot:462]:
```suggestion
msgid "Run Arch-Update"
```

File: po/fr.po [hunk 2]
```diff
@@ -491,3 +491,11 @@ msgstr ""
 #, sh-format
 msgid "The '${config_file}' configuration file has been generated"
 msgstr "Le fichier de configuration '${config_file}' a été généré"
+
+#: src/script/arch-update-tray.py:111
+msgid "Update"
```
Comment (Antiz96) [inline po/fr.po:496]:
```suggestion
msgid "Run Arch-Update"
```

File: src/script/arch-update-tray.py [hunk 3]
```diff
@@ -83,6 +92,18 @@ def __init__(self, statefile):
         self.tray.setVisible(True)
         self.tray.activated.connect(self.update)
 
+        # Menu
+        menu = QMenu()
+        menu_launch = QAction(_("Update"))
```
Comment (Antiz96) [inline src/script/arch-update-tray.py:97]:
```suggestion
        menu_launch = QAction(_("Run Arch-Update"))
```

**Comment by trigg** [comment]
Makes sense, done.

**Comment by trigg** [comment]
Good point too, should be easier to understand

**Comment by Antiz96** [comment]
I'm currently testing this. Will merge if everything works as intended :)

**Comment by Antiz96** [comment]
The right click menu works perfectly and is really great! :smile: 
However, I can't make the FR translation work for some reason... Running the script and the man pages with `LANGUAGE=fr` works as intended but the it does not for the systray. :thinking: 

Running `LANG=fr arch-update-tray` reports `Gtk-WARNING **: Locale not supported by C library. Using the fallback 'C' locale.`, despite the fact `fr_FR.utf-8` being enabled in my test env. Am I missing something? Any idea?

**Comment by Antiz96** [comment]
Well, I even set my test system in french completely and the menu still is in english. So there definitely something missing, either on my side or in this PR ^^ I'm investigating.

**Comment by trigg** [comment]
Double check you're installing a newer .mo file. I'll have another look shortly, but it should be automatically accounting for prefix using xdg

**Comment by Antiz96** [comment]
> Double check you're installing a newer .mo file. I'll have another look shortly, but it should be automatically accounting for prefix using xdg

I can confirm

**Comment by Antiz96** [comment]
Ok found the issue! The python script is looking for the .mo file in the `/usr` prefix, while I'm installing with the default `/usr/local` prefix in my test env.
The main script [looks both at `/usr` and `/usr/local`](https://github.com/Antiz96/arch-update/blob/main/src/script/arch-update.sh#L33-L35). Can something similar be achieved?

**Comment by Antiz96** [comment]
(Sorry for mistakenly closing the PR... I miss clicked once again :sweat_smile:)

**Comment by trigg** [comment]
> Can something similar be achieved?

It absolutely can, give me a while to look into this

**Comment by trigg** [comment]
I might add, it specifically checks for a French translation, as that is almost guaranteed in this project,

**Comment by Antiz96** [comment]
> I might add, it specifically checks for a French translation, as that is almost guaranteed in this project,

Oh! Yeah okay, smart! I missed the logic here. Ignore my previous comment.

**Comment by trigg** [comment]
fallback is set, and I'd be very worried about systems with neither XDG_DATA_DIRS and XDG_DATA_HOME.

In my case it covers
`/home/trigg/.wayfire/share:/home/trigg/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share`

**Comment by Antiz96** [comment]
BTW, I'll also update the main bash script to look for XDG_DATA_DIRS/XDG_DATA_HOME for the .mo file (in addition of /usr | /usr/local). I think this is the last thing where the script assumes `/usr`  or `/usr/local` without looking into XDG dirs as well so the script should be all clean on that front after that \o/

**Comment by Antiz96** [comment]
> fallback is set, and I'd be very worried about systems with neither XDG_DATA_DIRS and XDG_DATA_HOME.

Well, I have neither defined :angel:

**Comment by trigg** [comment]
Impressive! One moment then

**Comment by Antiz96** [comment]
It should look for XDG_DATA_HOME, then XDG_DATA_DIRS, than /usr/local, than /usr.

**Comment by trigg** [comment]
It will now iterate above XDG dirs, but ensure that the two expected directories are on the end of the list.

File: src/script/arch-update-tray.py [hunk 4]
```diff
@@ -22,6 +23,26 @@
     log.error("Statefile does not exist: %s", STATE_FILE)
     sys.exit(1)
 
+# Find translations
+paths = []
+if 'XDG_DATA_DIRS' in os.environ:
```
Comment (Antiz96) [inline src/script/arch-update-tray.py:28]: Correct me if I'm wrong, but there's a flaw in the logic here.

If someone has XDG_DATA_DIRS defined to a custom dir but still used the default `/usr/local` prefix at install (or `/usr` for the AUR package), the translation will be reported "not found".
I think we should use the same logic as for the Desktop file, like "if XDG_DATA_DIRS is defined, look into it. If not found there, than if XDG_DATA_HOME is defined, look into it. If not found there [...]". Right?

**Comment by trigg** [comment]
paths starts empty.

If XDG_DATA_DIRS is set, it's appended

if XDG_DATA_HOME is set, it's appended

regardless of both, `/usr` and `/usr/local` are appended.

We then iterate over this list until the first positive match comes up, and use that one.

In your case it'll end on the 2nd element `/usr/local`, in my case it'll end up on 5th element `/usr`. Either way we should both be covered by this now

**Comment by Antiz96** [comment]
Uh, I didn't catch that `path` was an array with all the potential path appended to it. My bad!

**Comment by trigg** [comment]
It was placed after to allow user to override it intentionally

**Comment by Antiz96** [comment]
Still not working on my side for some reason (both with the `/usr/local` and `/usr` prefixes) :shrug: 

I gotta go right now, I'll make some more testing/investigation either this evening or tomorrow :)
Thanks once again for all the work!

**Comment by Antiz96** [comment]
I made some more tests and I can't make the translation works with the current code... I tried to play a bit with the script but I can't seem to find the issue.

I can confirm the translation (.mo) file works though, as replacing the whole current "# Find translations" part of the script by the following makes the translation works when using the `/usr` prefix:
```
t = gettext.translation('Arch-Update', fallback=True)
_ = t.gettext
```

My guess is that there's something wrong with the [`localedir` definition](https://github.com/Antiz96/arch-update/pull/165/files#diff-9ca49406add1ba4740c95fbd61f6485ba6f966e5130c7038f5812fb5c1448987R38), as if the script was not looking into it/using it. But that's just an assumption... 

@trigg any idea?

I'll keep investigating on my side as well.

**Comment by trigg** [comment]
My apologies, something else is taking all my time recently. I'll look into this again ASAP

**Comment by Antiz96** [comment]
No problem :) Thanks!

**Comment by trigg** [comment]
It is concerning to me that my setup works before and after this edit.

This post-fixes 'locale' to the path, as that is what the python docs state is expected

**Comment by Antiz96** [comment]
Works as expected on my side now :tada:

**Comment by Antiz96** [comment]
@trigg How does that look? https://github.com/Antiz96/arch-update/pull/165/commits/5f4209bcda29d945529a19497f9efdc299b89bf8

**Comment by Antiz96** [comment]
Woops, gotta get rid of the `.split(":")` part right?

**Comment by trigg** [comment]
I've merged the parts into one path and appended that

**Comment by trigg** [comment]
the `extend` adds each segment of that path as if it was an entire path alone

**Review by Antiz96 — APPROVED** [review]
LGTM, thanks! :tada:

**Pull request merged**